src/crimson/osd/osd.cc

```diff
@@ -1178,8 +1183,10 @@ seastar::future<> OSD::committed_osd_maps(
       }
       if (should_restart()) {
         return restart();
+      } else if (!pg_shard_manager.is_stopping()) {
+	return get_shard_services().osdmap_subscribe(osdmap->get_epoch() + 1, false);
```

Comment:
This will probably work. IIUC, this is part of `start_waiting_for_healthy()` missing logic. Perhaps this is a good timing to introduce this in Crimson. What do you think?

Note: `is_waiting_for_healthy` is implemented but not used _yet_